### PR TITLE
[askeladd] Round-2: Larger grad-clip threshold (1.0 → 5.0, single delta)

### PR DESCRIPTION
**Pull request by morganmcg1**
# Round-2 Assignment: Larger gradient clip threshold (single delta)

## Hypothesis

Frieren's PR #42 (squared_rel_l2) gave us a critical diagnostic:
**`train/grad/clipped` was 1.0 at every single step** (23,458/23,458 steps),
regardless of loss form (MSE or squared rel-L2). The default `--grad-clip-norm 1.0`
is binding for every optimizer step in this configuration.

Late-training median pre-clip grad_norm is ~4 — so the 1.0 clip is throwing away
~75% of the gradient magnitude in steady state. This is signal we are paying GPU
hours to compute and then discarding.

**Single delta:** raise `--grad-clip-norm` from 1.0 to 5.0 on the new compiled
baseline (PR #40). No other changes.

If this helps, the natural follow-up is `lr=1e-4` + `grad-clip 5.0` (more steps
× larger steps), but let's first isolate the clip-only effect.

## Why this isn't redundant with what's running

- **alphonse #46** = RFF + compile, default clip 1.0
- **fern #43** = multi-scale RFF, default clip 1.0
- **edward #44** = cosine EMA + RFF, default clip 1.0
- **frieren #42** = squared_rel_l2 + compile, default clip 1.0 (still binding)
- **nezuko #35**, **tanjiro #39**, **thorfinn #47** = various, all default clip 1.0

No one is touching grad-clip. This is an orphaned single-delta lever and the
grad-norm telemetry strongly motivates it.

## Run command

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent askeladd \
  --wandb-name "askeladd/round2-grad-clip-5-512d" \
  --wandb-group "tay-round2-grad-clip-sweep" \
  --volume-loss-weight 2.0 \
  --batch-size 4 \
  --validation-every 1 \
  --lr 5e-5 --weight-decay 5e-4 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.9995 \
  --grad-clip-norm 5.0 \
  --gradient-log-every 100 --weight-log-every 100 \
  --no-log-gradient-histograms
```

**No `--no-compile-model`** — PR #40 fixed the bug. Compiled is the default.

## Optional bonus arm

If your run finishes and there's budget left, add a second arm with
`--grad-clip-norm 0` (clip off entirely) so we can see if any clipping at all
is appropriate. Same `--wandb-group tay-round2-grad-clip-sweep` so they group
together for comparison.

## Diagnostics to report

- `train/grad/clipped` rate over training — should be much lower than 1.0,
  ideally only firing during the early gradient-spike phase
- `train/grad/global_norm_pre_clip` quantile distribution at end of training
- Whether validation is more or less stable epoch-to-epoch
- **Critical:** whether any single training step explodes (loss → NaN / grad → inf)

## What to report

After training completes, post a comment with all `test_primary/*` metrics:

| Metric | This run | PR #40 (SOTA) |
|---|---:|---:|
| `abupt` | ? | 17.25 |
| `surface_pressure` | ? | 10.92 |
| `wall_shear` | ? | 18.33 |
| `volume_pressure` | ? | 14.71 |
| `tau_x` | ? | 15.73 |
| `tau_y` | ? | 21.80 |
| `tau_z` | ? | 23.07 |

Plus: W&B rank-0 run ID, wall-clock time, epochs reached, best val_abupt,
final-epoch grad-norm distribution.

## Failure mode to watch

The 1.0 clip threshold may be **load-bearing** — i.e., it could be implicit
regularization that prevents the model from collapsing into a local minimum.
If your run shows training loss spikes or val_abupt jumps around significantly,
that's the signal that clipping was helping. Drop the run if it diverges (val
> 100 sustained, or loss → NaN); we'll learn from that signal too.

## Current baseline to beat

| Metric | tay best (PR #40) | yi best | AB-UPT |
|---|---:|---:|---:|
| `test_primary/abupt_axis_mean_rel_l2_pct` | **17.25** | 15.82 | — |
| `test_primary/surface_pressure_rel_l2_pct` | **10.92** | 9.99 | 3.82 |
| `test_primary/wall_shear_rel_l2_pct` | **18.33** | 16.60 | 7.29 |
| `test_primary/volume_pressure_rel_l2_pct` | **14.71** | 14.21 | 6.08 |
| `test_primary/wall_shear_x_rel_l2_pct` | **15.73** | 14.27 | 5.35 |
| `test_primary/wall_shear_y_rel_l2_pct` | **21.80** | 19.49 | 3.65 |
| `test_primary/wall_shear_z_rel_l2_pct` | **23.07** | 21.12 | 3.63 |

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #49 because ADVISOR: Closing as a clear regression vs the new SOTA.

## Result vs current SOTA

The new SOTA (tanjiro arm B `vnb7oheo`, Lion lr=5e-5/wd=5e-4) is **test_abupt 11.30**.

| Metric | This run (PR #49) | New SOTA (Lion 5e-5) | Δ |
|---|---:|---:|---:|
| `test_primary/abupt_axis_mean_rel_l2_pct` | **15.232** | 11.303 | **+34.8% worse** |
| `val_primary/abupt_axis_mean_rel_l2_pct` | 14.219 | 10.096 | +40.8% worse |
| `surface_pressure` | 9.338 | 6.216 | +50.2% worse |
| `wall_shear` | 16.073 | 11.315 | +42.0% worse |
| `volume_pressure` | 13.796 | 12.755 | +8.2% worse |
| `tau_y` | 19.105 | 13.831 | +38.1% worse |
| `tau_z` | 20.067 | 14.147 | +41.9% worse |

W&B run: `x09udzj3`, group `tay-round2-grad-clip-sweep`, 284.9 min, best_epoch=16. Run finished cleanly.

## Diagnosis: clip-5 + AdamW under-trains vs Lion

The hypothesis was sound — at clip 1.0 we discard ~75% of the gradient signal late in training. Raising to clip 5.0 with AdamW does what it should: every-step grad pass-through, more loss reduction per epoch. But the experimental landscape shifted under us:

1. **Lion at lr=5e-5/wd=5e-4 sidesteps the issue entirely.** Sign-based updates make pre-clip grad-norm irrelevant — Lion's per-step movement is bounded by `lr` directly, not by gradient magnitude. PR #39's win and now `vnb7oheo`'s SOTA both come from this same property. Larger clip on AdamW recovers some of the discarded magnitude but does not match Lion's normalized-per-channel update geometry.

2. **Beats PR #46 14.55 only marginally** (+4.7%). On its own this would have been a "send back, try lr=1e-4 + clip 5.0" decision. Against 11.30 it's not competitive.

3. **No interaction with the new SOTA approach.** Lion + clip 5.0 is ill-defined (Lion is not magnitude-bounded), so this lever doesn't compose forward into the Lion arms (#50/#51/#52/#54).

## What's preserved

The diagnostic from frieren's PR #42 is still valuable: AdamW with default clip 1.0 throws away ~75% of late-stage gradient signal. We just chose Lion instead.

## What's next

You'll be reassigned a fresh round-3 hypothesis under the Lion lr=5e-5 SOTA. Closing this PR; the branch and W&B run are preserved for reference..
